src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSCopy.java

```diff
@@ -68,8 +62,16 @@ public void run(ActionContext context) throws IOException {
       collector.getOrThrowException();
       return;
     }
-    StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccount(),
-                                                       isServiceAccountFilePath, config.readTimeout);
+    StorageClient storageClient = null;
+    try {
+      storageClient = StorageClient.create(config.getProject(), config.getServiceAccount(), isServiceAccountFilePath,
+        config.readTimeout);
+    } catch (IOException e) {
```

Comment:
this should return `StorageException` as well, catch that and handle accordingly.